### DISCUSSION TITLE
`astro:env` allowing numbers in variable names

### DISCUSSION THREAD

**Discussion by NuroDev** [withastro/roadmap#972] [Proposal]
### Body

# Summary

Currently if your `astro:env` environment variable schema includes any numbers in the name it will throw the following error:
```bash
[config] Astro found issue(s) with your configuration:
  ! experimental.env.schema.S3_ACCESS_KEY_ID  A valid variable name can only contain uppercase letters and underscores..
```

# Background & Motivation

I only noticed this when trying to set up `astro:env` to have strict / typed environment variables to access my S3 bucket, which of course requires a number of variables (Hostname, bucket name, access key ID & secret). Of course these variables could just be renamed something different but ideally it would make the most sense to include some kind of "S3" in the variable name, but currently that is not allowed.

# Goals

 - Update [the `astro:env` regex pattern](https://github.com/withastro/astro/blob/main/packages/astro/src/env/schema.ts#L84) to also allow numbers as well as upper case characters.
 - (Optional) Not allow variables starting with a number as that could cause issues?

**Comment by delucis** [comment]
Hi @NuroDev! This seems like a bug to me more than a feature request. I’d recommend filing a bug in the main Astro repo: https://github.com/withastro/astro/issues/new?assignees=&labels=&projects=&template=---01-bug-report.yml
